autotest/gdrivers/jp2kak.py

```diff
@@ -506,3 +507,294 @@ def test_jp2kak_image_origin_not_zero():
     ds = gdal.Open('data/jpeg2000/byte_image_origin_not_zero.jp2')
     assert ds.GetRasterBand(1).Checksum() == 4672
     assert ds.GetRasterBand(1).ReadRaster(0,0,20,20,10,10) is not None
+
+###############################################################################
+# Test multiple RATE parameters
+
+def test_jp2openjpeg_test_multi_rate():
+    src_ds = gdal.Open('../gcore/data/stefan_full_rgba.tif')
+    gdaltest.jp2kak_drv.CreateCopy('/vsimem/jp2kak_22.jp2', src_ds, options=['RATE=1.5,2,3,4', 'LAYERS=4'])
+    node = gdal.GetJPEG2000Structure('/vsimem/jp2kak_22.jp2', ['ALL=YES'])
+
+    arr = []
+    find_elements_with_name(node, "Field", "COM", arr)
+
+    # Approximation of [4556, 6075, 9112, 12150] = floor(dfPixelsTotal * currentBitRate * 0.125F)
```

Comment:
woo, thanks for having tried that! I didn't image something as evolved as what you tried . Unfortunately all those tests based on the content of the COM marker are too fragile. They break with the Kakadu version I've at hand. I would just suggest getting the COD.SGcod_NumLayers layers and checking it. That's not very in-depth, but better than nothing.